roles/cloud_connector/tasks/cleanup_receptor.yml

```diff
@@ -0,0 +1,61 @@
+---
+- name: Find receptor configs
+  find:
+    paths: /etc/receptor
+    file_type: directory
+  register: receptor_configs
+
+- name: Identify Satellite ID from Satellite
+  ansible.builtin.uri:
+    url: "https://{{ satellite_url }}/api/settings?search=name%20%3D%20instance_id"
+    user: "{{ satellite_username }}"
+    password: "{{ satellite_password }}"
+    return_content: yes
+    validate_certs: yes
+    force_basic_auth: yes
+  register: satellite_uuid_json
+
+- include_tasks: delete_source.yaml
+  loop: "{{ receptor_configs['files'] }}"
+
+ name: Check if foreman-maintain installed
+  ansible.builtin.package_facts:
+    manager: auto
+
+- name: Collect service facts
+  ansible.builtin.service_facts:
+
+- name: Stop any receptor services
+  ansible.builtin.service:
+    state: stopped
+    enabled: false
+    name: "{{ item.name }}"
+  loop: "{{ ansible_facts.services.values() | selectattr('name', 'regex', 'receptor\\@.+.service') | list }}"
+
+- when: (['receptor', 'python3-receptor-satellite', 'ansiblerole-satellite-receptor-installer'] | difference(ansible_facts.packages.keys()) | list | length) == 0
```

Comment:
Maybe this is unnecessary but keeps from having to do lock and unlock on foreman-maintain.